metabase/metabase
# fix echarts timezone issues

Closes https://github.com/metabase/metabase/issues/40204
Closes https://github.com/metabase/metabase/issues/39322

### Description

ECharts selects ticks based on the browser timezone by default and there is no way to specify a custom timezone for them. Because of that, when the dataset timezone is vastly different from the browser timezone and formatting omits time, the tick can be placed at yesterday/tomorrow of the dataset timezone which is in `results_timezone` prop.

Since there is no way to make ECharts selecting ticks in the needed timezone, we have to shift dates in the dataset so it looks like ticks and the data are in the same timezone. ECharts can be forced to pick ticks values in UTC by setting `useUTC: true` which I used in the PR. In addition to it, I shift time series x-values like they are in UTC instead of the `results_timezone`.

### How to verify

Check time series charts show ticks in the `results_timezone` no matter what is the browser timezone.
It is convenient to test it with native queries that return a single row like `SELECT CAST('2022-01-01 00:00:00-03:00' AS TIMESTAMP WITH TIME ZONE) AS x, 10 y`.

The browser timezone can be changed in Chrome devtools Sensors tab:
<img width="417" alt="Screenshot 2024-03-27 at 8 14 52 PM" src="https://github.com/metabase/metabase/assets/14301985/5d150bcb-d61d-4046-8586-b235eb2c84f7">

### Demo

#### Ticks timezone Loom
https://www.loom.com/share/567fb2996bb14a6a9ca65cc2ad5d441a?sid=3267a799-6ea9-40a4-8491-cf007efa8596

#### Timeline events

Before
<img width="1039" alt="Screenshot 2024-03-27 at 8 10 25 PM" src="https://github.com/metabase/metabase/assets/14301985/d9b406e0-80ef-427c-ad14-46aa3c922759">

After
<img width="1039" alt="Screenshot 2024-03-27 at 8 10 44 PM" src="https://github.com/metabase/metabase/assets/14301985/f53e66a5-a092-4144-9a6c-44d53cf06d41">

### Checklist
